ngmt/modules/td/_pham.py

```diff
@@ -0,0 +1,372 @@
+            accel /= 9.81
+
+        # Check unit of acceleration data if it is in g or m/s^2
+        if gyro_data_unit == "rad/s":
+            # Convert acceleration data from rad/s to deg/s (if not already is in deg/s)
+            gyro = np.rad2deg(gyro)
+
+        # Convert gyro data unit from deg/s to rad/s
+        gyro *= self.gyro_convert_unit
+
+        # Compute the variance of the moving window of gyro signal
+        gyro_vars = []
+
+        for i in range(3):
+            gyro_var = preprocessing.moving_var(data=gyro[:, i], window=sampling_freq_Hz)
+            gyro_vars.append(gyro_var)
+
+        gyro_var_1, gyro_var_2, gyro_var_3 = gyro_vars
+
+        # Threshold value for identifying periods where the variance is low
+        thr = 2 * 10**-4
+        
+        # Identify periods where the variance is below the threshold
+        self.locs = (gyro_var_1 <= thr) & (gyro_var_2 <= thr) & (gyro_var_3 <= thr)
+        
+        # Exclude the last 'sampling_freq_Hz' samples from the identified periods
+        self.locs[-sampling_freq_Hz:] = False
+        
+        # If no bias periods are found, return the original gyro values
+        if np.sum(self.locs) == 0:
+            gyro_out = gyro
+            return gyro_out
+        
+        # Find the start and end indices of the first bias period
+        locs_place = np.where(self.locs)[0]
+        location = np.array([[locs_place[0]], [locs_place[0] + sampling_freq_Hz - 1]])
+        
+        # Calculate the bias for each axis within the identified period
+        self.gyro_bias = np.mean(gyro[location[0, 0]:location[1, 0] + 1], axis=0)
+        
+        # Subtract the bias from the original gyro signal
+        gyro = gyro - self.gyro_bias
+
+        # Integrate x component of the gyro signal to get yaw angle (also convert gyro unit to deg/s)
+        self.yaw = scipy.integrate.cumtrapz(gyro[:, 0] / self.gyro_convert_unit, initial=0) / sampling_freq_Hz
+
+        # Find zero-crossings indices
+        self.index_zero_crossings = np.where(np.diff(np.sign(gyro[:, 0])))[0]  
+
+        # Calculate turns from yaw angle
+        self.turns_all = self.yaw[self.index_zero_crossings[1:]] - self.yaw[self.index_zero_crossings[:-1]]  
+        
+        # Marks hesitations in the signal
+        # Initialize an array to mark hesitations
+        hesitation_markers = np.zeros(len(self.turns_all))
+
+        # Loop through each index in the turns_all array
+        for i in range(len(self.turns_all)):
+            # Check if the absolute value of the turn angle at index i is greater than or equal to 10
+            if abs(self.turns_all[i]) >= 10:
+                # Loop to search for potential hesitations
+                for j in range(i + 1, len(self.turns_all)):
+                    # Check if the difference between current index and i exceeds 4, or if the time between zero crossings exceeds half a second
```

Comment:
I think the minimum turn duration and the maximum turn duration should also be parameters that are passed to the detection algorithm.

Reply:
The minimum turn duration (min_turn_duration_s) and the maximum turn duration (max_turn_duration_s) are now passed to the detection.